core/lib/types/src/kzg.rs

```diff
@@ -0,0 +1,315 @@
+        let opening_point = Bytes32::from_bytes(&opening_point).unwrap();
+
+        let (opening_proof, opening_value) =
+            KzgProof::compute_kzg_proof(&blob, &opening_point, &kzg_settings).unwrap();
+
+        let blob_proof =
+            KzgProof::compute_blob_kzg_proof(&blob, &kzg_commitment.to_bytes(), &kzg_settings)
+                .unwrap();
+
+        Self {
+            blob,
+            kzg_commitment,
+            opening_point,
+            opening_value,
+            opening_proof,
+            versioned_hash,
+            blob_proof,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::{Deserialize, Serialize};
+    use serde_with::serde_as;
+
+    use super::{KzgInfo, KzgProof};
+    use crate::{H256, U256};
+
+    #[serde_as]
+    #[derive(Debug, Serialize, Deserialize)]
+    struct ExepectedOutputs {
+        versioned_hash: H256,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        kzg_commitment: Vec<u8>,
+        opening_point: U256,
+        opening_value: U256,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        opening_proof: Vec<u8>,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        blob_proof: Vec<u8>,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        pubdata_commitment: Vec<u8>,
+    }
+
+    #[serde_as]
+    #[derive(Debug, Serialize, Deserialize)]
+    struct KzgTest {
+        #[serde_as(as = "serde_with::hex::Hex")]
+        pubdata: Vec<u8>,
+        expected_outputs: ExepectedOutputs,
+    }
+
+    #[test]
+    fn kzg_test() {
+        let zksync_home = std::env::var("ZKSYNC_HOME").unwrap_or_else(|_| ".".into());
+        let path = std::path::Path::new(&zksync_home).join("etc/kzg_tests/kzg_test_0.json");
+        let contents = std::fs::read_to_string(path).unwrap();
+        let kzg_test: KzgTest = serde_json::from_str(&contents).unwrap();
+
+        let kzg_info = KzgInfo::new(kzg_test.pubdata);
+
+        assert_eq!(
+            kzg_test.expected_outputs.versioned_hash,
+            H256::from_slice(kzg_info.versioned_hash.as_slice())
+        );
+
+        assert_eq!(
+            kzg_test.expected_outputs.kzg_commitment,
+            kzg_info.kzg_commitment.to_bytes().as_slice()
+        );
+
+        assert_eq!(
+            kzg_test.expected_outputs.opening_point,
+            U256::from_big_endian(kzg_info.opening_point.as_slice())
+        );
+
+        assert_eq!(
+            kzg_test.expected_outputs.opening_value,
+            U256::from_big_endian(kzg_info.opening_value.as_slice())
+        );
+
+        assert_eq!(
+            kzg_test.expected_outputs.opening_proof,
+            kzg_info.opening_proof.to_bytes().as_slice()
+        );
+
+        assert_eq!(
+            kzg_test.expected_outputs.blob_proof,
+            kzg_info.blob_proof.to_bytes().as_slice()
+        );
+
+        let encoded_info = kzg_info.to_bytes();
+        let kzg_info_2 = KzgInfo::from_slice(&encoded_info);
```

Comment:
maybe it would be possible to autoimplement `PartialEq, Eq` traits? For faster comparison

Reply:
i had to implement the traits since we dont want to compare them all